go-controller/pkg/ovn/master.go

```diff
@@ -485,11 +497,51 @@ func (oc *Controller) ensureNodeLogicalNetwork(nodeName string, hostsubnet *net.
 	return nil
 }
 
+// annotate the node with the subnet information assigned to node's logical switch. the
+// new format of the annotation is:
+//
+// k8s.ovn.org/node-subnets: {
+//	  "default": "192.168.2.1",
+// }
+func (oc *Controller) addNodeAnnotations(node *kapi.Node, subnet string, replace bool) error {
+	// nothing to do if the node already has the annotation key
+	_, ok := node.Annotations[OvnHostSubnet]
+	if ok {
+		return nil
+	}
+
+	bytes, err := json.Marshal(map[string]string{
+		"default": subnet,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal node %q annotation for subnet %s",
+			node.Name, subnet)
+	}
+	nodeAnnotations := make(map[string]interface{})
+	if replace {
+		nodeAnnotations[OvnHostSubnetLegacy] = nil
+	}
+	nodeAnnotations[OvnHostSubnet] = string(bytes)
+	err = oc.kube.SetAnnotationsOnNode(node, nodeAnnotations)
+	if err != nil {
+		return fmt.Errorf("failed to set node annotation %q on existing node %s to %q: %v",
+			OvnHostSubnet, node.Name, subnet, err)
+	}
+	return nil
+}
+
 func (oc *Controller) addNode(node *kapi.Node) (hostsubnet *net.IPNet, err error) {
 	oc.clearInitialNodeNetworkUnavailableCondition(node)
 
 	hostsubnet, _ = parseNodeHostSubnet(node)
 	if hostsubnet != nil {
+		// Update the node's annotation to use the new annotation key and remove the
+		// old annotation key (this is fine since this key is used only by the
+		// master and worker nodes don't use the node-subnet key).
+		err = oc.addNodeAnnotations(node, hostsubnet.String(), true)
```

Comment:
The parenthetical only makes sense in the context of the fact that we're *changing* the annotation, and so pretty soon it will be completely unnecessary. It makes sense to explain that in the PR (as you already did) but I don't think there's any reason to put it in the code.

Reply:
Agree. Will remove the text within parenthesis.